### ISSUE TITLE
REQUEST: Repository maintenance on opentelemetry-network-build-tools

### ISSUE THREAD

**Issue by yonch** [open-telemetry/community#2387]
### Affected Repository

[https://github.com/open-telemetry/opentelemetry-network-build-tools](https://github.com/open-telemetry/opentelemetry-network-build-tools)

edit: caption of link

### Requested changes

1. As a maintainer, I'd like to be able to merge my own PRs without another maintainer's review.
2. The repository should have secrets configured: DOCKER_USERNAME and DOCKER_PASSWORD
3. The repository should have vars configured: DOCKER_REGISTRY and DOCKER_NAMESPACE
 
### Purpose

This repository is the source needed to build the `build-env` container for opentelemetry-network. The project uses a build environment to allow contributors to work on the exact same version of dependencies, so contributions compile reliably.

I'm adding GitHub Actions to build and release this `build-env`. This requires iteration on the main branch and the PR rules make this unwieldy...

### Expected Duration

2 weeks for the PR merge.

secrets and vars should remain permanently.

### Repository Maintainers

- @open-telemetry/network-maintainers

**Comment by trask** [comment]
> I'm adding GitHub Actions to build and release this `build-env`. This requires iteration on the main branch and the PR rules make this unwieldy...

I usually test GitHub Actions on my fork.

I know @austinlparker uses https://github.com/nektos/act to test GitHub Actions locally.

would either of these approaches alleviate the burden for you?

**Comment by yonch** [comment]
Thanks for the pointers!

I'm close to the point where I need to test with large builders (the build needs ~50GB of disk space) and don't have access to those on my private account..

I'll get as close as I can with act but if I can't merge into this repo it will likely push this effort beyond Kubecon (I'm working on this on my own time). Would appreciate the merge exception if possible..

**Comment by yonch** [comment]
All right, the `act` recommendation was great, thank you @trask !

The opentelemetry-network-build-tools PR is now blocked on PR merges.

Also can someone chime in if DOCKER_* is already available in the repo or it needs to be configured?

**Comment by trask** [comment]
> Also can someone chime in if DOCKER_* is already available in the repo or it needs to be configured?

cc @tigrannajaryan

**Comment by tigrannajaryan** [comment]
1. https://hub.docker.com/repository/docker/otel/opentelemetry-network-build-tools docker hub repo is created.
2. DOCKER_USERNAME and DOCKER_PASSWORD env variables added to https://github.com/open-telemetry/opentelemetry-network-build-tools and should be available to actions.
3. The user has permissions to Read&Write the docker repo.

**Comment by yonch** [comment]
Great, thank you @tigrannajaryan !

@trask can you please override the review requirement for maintainers for the next week?

**Comment by trask** [comment]
> @trask can you please override the review requirement for maintainers for the next week?

hey @yonch, I'm not comfortable overriding that requirement on my own and would need to check first with the Technical Committee / Governance Committee.

Is there a time zone issue that is preventing you from being able to get timely approvals? cc @open-telemetry/network-approvers

**Comment by yonch** [comment]
@trask it's not the timely approvals, just that I know folks are busy right now and wanted to avoid creating load. 

But @bjandras said he could help so we can close this for now. Thank you for the help!
